yt_dlp/extractor/youtube.py

```diff
@@ -2262,8 +2265,11 @@ def feed_entry(name):
             'categories': [category] if category else None,
             'tags': keywords,
             'is_live': is_live,
+            'is_upcoming': is_upcoming,
             'playable_in_embed': playability_status.get('playableInEmbed'),
             'was_live': video_details.get('isLiveContent'),
+            'live_starttime': try_get(live_broadcast_details, lambda x: x['startTimestamp'], dict),
+            'live_endtime': try_get(live_broadcast_details, lambda x: x['endTimestamp'], dict),
```

Comment:
Set starttime to `timestamp` and "endtime - starttime" to duration for upcoming videos. There is no need for the new fields

Reply:
If/when changing `live_starttime` to `timestamp`, convert it to an integer timestamp of seconds like Twitch instead of ISO 8601